client/src/main/scala/org/http4s/client/middleware/UnixSocket.scala

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2014 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.client.middleware
+
+import org.http4s.client.Client
+import org.http4s.Request
+import cats.effect.kernel.MonadCancelThrow
+
+/** Middleware to direct all requests to the provided `UnixSocketAddress` */
+object UnixSocket {
+  def apply[F[_]: MonadCancelThrow](address: fs2.io.net.unixsocket.UnixSocketAddress)(
```

Comment:
Is this necessary because the current http4s model can't specify unix socket addresses as destinations ?

Reply:
UnixSockets doing http is basically unspecified. This was intended as a convenient way to automatically redirect all requests to the UnixSocket.